src/initgeneration/initgeneration.go

```diff
@@ -26,27 +26,29 @@ type InitGenerator struct {
 	apiReader     client.Reader
 	namespace     string
 	canWatchNodes bool
+	dynakubeQuery *query.DynakubeQuery
 }
 
 type nodeInfo struct {
 	nodes   []corev1.Node
 	imNodes map[string]string
 }
 
-func NewInitGenerator(client client.Client, apiReader client.Reader, ns string) *InitGenerator {
+func NewInitGenerator(client client.Client, apiReader client.Reader, namespace string) *InitGenerator {
 	return &InitGenerator{
-		client:    client,
-		apiReader: apiReader,
-		namespace: ns,
+		client:        client,
+		apiReader:     apiReader,
+		namespace:     namespace,
+		dynakubeQuery: query.NewDynakubeQuery(client, namespace),
 	}
 }
 
 // GenerateForNamespace creates the init secret for namespace while only having the name of the corresponding dynakube
 // Used by the podInjection webhook in case the namespace lacks the init secret.
-func (g *InitGenerator) GenerateForNamespace(ctx context.Context, dk dynatracev1beta1.DynaKube, targetNs string) (bool, error) {
```

Comment:
why remove `ctx`??

I see its not used, but I also see that it could be (here: https://github.com/Dynatrace/dynatrace-operator/pull/806/files#diff-23aafca6480c6cd32914b1b7b9691cbf3acfe3d7ec384c2926b56ebd698c347aR133)

Reply:
refactored code to hell and back